### PR TITLE
[MU4] App menu fix

### PR DESCRIPTION
**Pull request by Eism**
Resolves: 
#8109
#8486

### CODE REVIEW

File: src/appshell/view/appmenumodel.cpp [hunk 1]
```diff
@@ -306,8 +306,11 @@ MenuItemList AppMenuModel::recentScores() const
     MenuItemList items;
     MetaList recentScores = userScoresService()->recentScoreList().val;
 
+    int index = 0;
     for (const Meta& meta : recentScores) {
-        MenuItem item = uiactionsRegister()->action("file-open");
+        MenuItem item;
+        item.id = QString::fromStdString(item.code) + QString::number(index++);
```
Comment (RomanPudashkin) [inline src/appshell/view/appmenumodel.cpp:312]: This code is duplicated. Better to move it to the static function

File: src/appshell/view/notationstatusbarmodel.cpp [hunk 2]
```diff
@@ -314,9 +325,9 @@ void NotationStatusBarModel::zoomOut()
     dispatch("zoomout");
 }
 
-void NotationStatusBarModel::handleAction(const QString& actionCode)
+void NotationStatusBarModel::handleMenuItem(const QString& itemId)
 {
-    dispatch(codeFromQString(actionCode));
+    dispatch(codeFromQString(itemId));
```
Comment (RomanPudashkin) [inline src/appshell/view/notationstatusbarmodel.cpp:330]: I believe you need to explicitly find the item by itemId and get actionCode from it

File: src/appshell/qml/platform/mac/Main.qml [hunk 3]
```diff
@@ -77,7 +77,7 @@ AppWindow {
         menuItem.checked = itemInfo.checked
         menuItem.checkable = itemInfo.checkable
         menuItem.shortcut = itemInfo.shortcut
-        menuItem.actionCode = itemInfo.code
+        menuItem.id = itemInfo.id
```
Comment (RomanPudashkin) [inline src/appshell/qml/platform/mac/Main.qml:80]: Please, move this line above menuItem.text

File: src/appshell/view/appmenumodel.cpp [hunk 4]
```diff
@@ -483,8 +486,10 @@ MenuItemList AppMenuModel::workspacesItems() const
         return workspace1->name() < workspace2->name();
     });
 
+    int index = 0;
     for (const IWorkspacePtr& workspace : workspaces) {
         MenuItem item = uiactionsRegister()->action("select-workspace");
+        item.id = QString::fromStdString(item.code) + QString::number(index);
```
Comment (RomanPudashkin) [inline src/appshell/view/appmenumodel.cpp:492]: index always will be 0

Better to move it inside for statement:

for (int i = 0; i < m_workspaces.size(); ++i) {
IWorkspacePtr& workspace = workspaces[i];
item.id = makeId(item.code, i);
...
}

File: src/notation/view/notationpaintview.cpp [hunk 5]
```diff
@@ -332,9 +332,9 @@ void NotationPaintView::hideContextMenu()
     emit hideContextMenuRequested();
 }
 
-void NotationPaintView::handleAction(const QString& actionCode)
+void NotationPaintView::handleMenuItem(const QString& itemId)
 {
-    dispatcher()->dispatch(actionCode.toStdString());
+    dispatcher()->dispatch(itemId.toStdString());
```
Comment (RomanPudashkin) [inline src/notation/view/notationpaintview.cpp:337]: I believe you need to explicitly find the item by itemId and get actionCode from it

File: src/notation/view/notationtoolbarmodel.cpp [hunk 6]
```diff
@@ -89,9 +91,9 @@ void NotationToolBarModel::load()
     });
 }
 
-void NotationToolBarModel::handleAction(const QString& actionCode)
+void NotationToolBarModel::handleMenuItem(const QString& itemId)
 {
-    dispatcher()->dispatch(actions::codeFromQString(actionCode));
+    dispatcher()->dispatch(actions::codeFromQString(itemId));
```
Comment (RomanPudashkin) [inline src/notation/view/notationtoolbarmodel.cpp:96]: I believe you need to explicitly find the item by itemId and get actionCode from it

**Comment by vshalkevich** [comment]
Can`t be merged for now - still have issues which should be fixed
#8109
#8486

**Comment by vshalkevich** [comment]
Checked. Bugs fixed

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**